pg-core/src/java/org/pg/util/SocketTool.java

```diff
@@ -5,14 +5,20 @@
 import java.io.IOException;
 import java.net.Socket;
 import java.net.SocketAddress;
+import java.util.concurrent.Future;
+import java.util.concurrent.ExecutionException;
+import java.nio.channels.AsynchronousSocketChannel;
 import java.nio.channels.SocketChannel;
 
 public class SocketTool {
 
-    public static SocketChannel open (final SocketAddress address) {
+    public static AsynchronousSocketChannel open (final SocketAddress address) {
         try {
-            return SocketChannel.open(address);
-        } catch (IOException e) {
+            final AsynchronousSocketChannel ch = AsynchronousSocketChannel.open();
+            Future result = ch.connect(address);
+            result.get();
```

Comment:
Since there is no a timeout option for async channel, it looks like we need to `.get` by timeout passing an amount ms, for example:

`result.get(config.SOTimeout(), TimeUnits/MILLISECONDS)`

This will imitate a socket timeout behaviour.

Reply:
We can do that, sure. It might make sense to call it connect timeout instead?